### PR TITLE
use fastify-plugin instead of patching directly the metadata

### PR DESCRIPTION
**Pull request by EliphazBouye**
Use fastify-plugin instead of patching directly the metadata this pull request is made for fix issue #280 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -261,11 +262,15 @@ async function fastifyHttpProxy (fastify, opts) {
...
-module.exports.fastifyHttpProxy = fastifyHttpProxy
+})
+module.exports.default = fp(fastifyHttpProxy, {
+  fastify: '4.x',
+  name: '@fastify/http-proxy'
+})
+module.exports.fastifyHttpProxy = fp(fastifyHttpProxy, {
+  fastify: '4.x',
+  name: '@fastify/http-proxy'
+})
```
Comment (Fdawgs) [inline index.js:276]:
```suggestion
module.exports.default = fastifyHttpProxy
module.exports.fastifyHttpProxy = fastifyHttpProxy
```

Follows the style as the other official plugins, i.e. https://github.com/fastify/fastify-rate-limit/blob/master/index.js#L279-L284

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (EliphazBouye) [reply index.js]: Great thanks for your advice 😊 I have just made the following modification taking the example of the plugin

Comment (EliphazBouye) [reply index.js]: But I keep get some error on tests suite 🤔

File: index.js [hunk 2]
```diff
@@ -261,11 +262,10 @@ async function fastifyHttpProxy (fastify, opts) {
...
 
-fastifyHttpProxy[Symbol.for('plugin-meta')] = {
+module.exports = fp(fastifyHttpProxy, {
   fastify: '4.x',
-  name: '@fastify/http-proxy'
-}
-
-module.exports = fastifyHttpProxy
+  name: '@fastify/http-proxy',
+  encapsulate: true
```
Comment (Eomm) [inline index.js:268]: Omg, I didn't expect that but it makes sense

Good spot!

**Review by Eomm — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (EliphazBouye) [reply index.js:268]: Thanks 😊

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@Fdawgs ptal

**Review by Fdawgs — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Pull request merged**